src/seismometer/report/fairness.py

```diff
@@ -0,0 +1,554 @@
+    cohort_dict : dict[str, tuple[Any]]
+        collection of cohort groups to loop over
+    censor_threshold : int, optional
+        Limit at which a cohort group will be removed from the table if not enough observations are found,
+        by default 10.
+
+    Returns
+    -------
+    HTML
+        The HTML table for the fairness evaluation
+    """
+    fairness_groups = []
+    metric_groups = []
+    icon_groups = []
+
+    if fairness_ratio <= 0:
+        raise ValueError("Fairness ratio must be greater than 0")
+
+    if not cohort_dict:
+        raise ValueError("No cohorts provided for fairness evaluation")
+
+    for cohort_column in cohort_dict:
+        cohort_indices = []
+        cohort_values = []
+        for cohort_class in cohort_dict[cohort_column]:
+            cohort_indices.append((cohort_column, cohort_class))
+
+            cohort_filter = FilterRule.eq(cohort_column, cohort_class)
+            cohort_dataframe = cohort_filter.filter(dataframe)
+
+            index_value = {"Count": len(cohort_dataframe)}
+            metrics = metric_fn(cohort_dataframe, metric_list, **kwargs)
+            index_value.update(metrics)
+
+            cohort_values.append(index_value)
+        cohort_data = pd.DataFrame.from_records(
+            cohort_values, index=pd.MultiIndex.from_tuples(cohort_indices, names=["Cohort", "Class"])
+        )
+        cohort_ratios = cohort_data.div(cohort_data.loc[cohort_data["Count"].idxmax()], axis=1)
+
+        cohort_icons = cohort_ratios.drop("Count", axis=1).applymap(
+            lambda ratio: FairnessIcons.get_fairness_icon(ratio, fairness_ratio)
+        )
+        cohort_icons["Count"] = cohort_data["Count"]
+
+        fairness_groups.append(cohort_ratios)
+        icon_groups.append(cohort_icons)
+        metric_groups.append(cohort_data)
+
+    fairness_data = pd.concat(fairness_groups)
+    metric_data = pd.concat(metric_groups)
+    fairness_icons = pd.concat(icon_groups)
+
+    for cohort_column, cohort_class in metric_data.index:
+        if metric_data.loc[(cohort_column, cohort_class), "Count"] < censor_threshold:
+            fairness_icons.loc[(cohort_column, cohort_class), "Count"] = FairnessIcons.UNKNOWN.value
+            for metric in metric_list:
+                fairness_icons.loc[(cohort_column, cohort_class), metric] = FairnessIcons.UNKNOWN
+                metric_data.loc[(cohort_column, cohort_class), metric] = np.nan
+                fairness_data.loc[(cohort_column, cohort_class), metric] = np.nan
+
+    fairness_icons[metric_list] = fairness_icons[metric_list].applymap(
+        lambda x: x.value if x != FairnessIcons.UNKNOWN else "--"
+    )
+    fairness_icons[metric_list] = (
+        fairness_icons[metric_list]
+        + metric_data[metric_list].applymap(lambda x: f"  {x:.2f}  " if not np.isnan(x) else "")
+        + fairness_data[metric_list].applymap(lambda x: f"  ({x-1:.2%})  " if not (np.isnan(x) or x == 1.0) else "")
```

Comment:
moved to isfinite

Reply:
I also needed to fix the negation :-(

Reply:
I'm getting very nitpick, I know but can you revert the metric_data bit to nulling on nan but have the % be on finite? 
I do think showing the "value" of inf can be appropriate, just the inf% seemed odd.  Or do you specifically prefer hiding 'inf'

This seems fine to me:
![image](https://github.com/user-attachments/assets/939a49b3-ed30-49b9-9e36-82d0c634b9c6)

Reply:
are you okay with the -100% on the zero, its basically sensitivity is 0.001 for the default class and real 0 for the small class, otherwise I fixed the inf back in the last column.